### DISCUSSION TITLE
Xdebug on multiple projects simultaneously

### DISCUSSION THREAD

**Discussion by Censin** [beyondcode/herd-community#979] [Q&A]
I'm switching from Laravel Sail to Laravel Herd and it's been great. However, one area that I've struggled with is debugging my projects using xdebug. My issue is that I have multiple projects and can't debug them simultaneously.

For example I have a primary application and a secondary smaller application which is essentially just api endpoints. I use vscode and I want to be able to have my debugger listening for both projects. When I was using Laravel Sail this was possible due to each project having their own containerized instance of php and I could just listen to different ports. However, in Laravel Herd, since it is using the same php under the hood, I can only send on one port for all projects using that version of php.

My hacky workaround for now is that one project uses php 8.3 and one project uses php 8.2 and I can get the functionality I want. This isn't ideal for obvious reasons, namely that both projects should be running php 8.2 right now. It would be nice if Herd could do multiple instances of the same php version.

Does anyone have any thoughts or direction on how I could deal with this issue?

**Comment by adrum** [comment]
Your debugger VSCode or PHPStorm can only have one debug session listening on port 9003 at a time. 

This is more of a limitation of how Xdebug and the debuggers work. 

If the IDEs could share and reuse the debug process across multiple windows, that would likely allow for this.

**Reply by Censin** [reply]
Yeah it would be nice if VSCode could share that process across instances. I have my php 8.2 configured to use port 9003 for Xdebug and my php 8.3 using port 9004 so my debuggers can listen on separate ports.
